### PR TITLE
Allow to have Redis on cluster mode

### PR DESCRIPTION
**Pull request by fabianaferreira**
#### What is the purpose of this pull request?
To allow Redis client to connect to a Redis on cluster mode instead of the local cluster caches.

#### What problem is this solving?
We're moving to shut down our caches per cluster and unify them on an Elasticache. However, our Redis client implementation does not support cluster mode. Therefore, it's necessary to change its implementation to enable this different way of connecting to the central Redis cluster.

#### How should this be manually tested?
We need to use it on Colossus by tagging a beta version and routing the requests from a workspace/account to the cluster that is currently using and validating the Elasticache connection.

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: redis/cache.go [hunk 1]
```diff
@@ -181,6 +198,17 @@ func (r *redisC) remoteKey(key string) (string, error) {
 }
 
 func (r *redisC) doCmd(cmd string, args ...interface{}) (interface{}, error) {
+	if r.cluster != nil {
+		// If it doesn't have the pool, then we're on cluster mode
```
Comment (MMartyn) [inline redis/cache.go:202]:
```suggestion
```

**Comment by raildo** [comment]
At first look, it feels a bit odd to me that we're copying a lot of Golang libs for internal packages. Do we intend to maintain this libs such as gin, httpcache and others?

**Review by MMartyn — APPROVED** [review]
LGTM - I would remove that comment as it isn't valid any longer

**Comment by raildo** [comment]
> 

I mean, I know that it should be just a "thin layer" on this repo, I just would like to know how that's gonna be maintained in long term, deprecation, releases, if we're choosing to provide this kind of wrapper for external libs. Do we have this documented somewhere else?

**Comment by fabianaferreira** [comment]
> we're copying a lot of Golang libs for internal packages. Do we intend to maintain this libs such as gin, httpcache and others?

These are related to running `go mod vendor`, which keeps a "local" copy of the dependencies, to avoid new versions from being used, unless we manually update them.

**Review by raildo — APPROVED** [review]

**Pull request merged**